components/net/fetch/request.rs

```diff
@@ -553,15 +552,15 @@ fn http_network_or_cache_fetch(request: Rc<RefCell<Request>>,
 
     // Step 1
     let http_request = if request_has_no_window &&
-        request.borrow().redirect_mode != RedirectMode::Follow {
+        request.redirect_mode != RedirectMode::Follow {
         request.clone()
     } else {
-        Rc::new(RefCell::new(request.borrow().clone()))
+        request.clone()
```

Comment:
Wait, sorry, this conditional check isn't even needed at all if both branches result in `request.clone()`.